### PR TITLE
Support internally tagged enums

### PR DESCRIPTION
**Pull request by trxcllnt**
This PR makes `serde-env` support internally-tagged enums.

Serde calls `Deserializer::deserialize_any` on internally-tagged enums. When that happens, the `serde-env` node's value is an empty string. Detect this and delegate to `Deserializer::deserialize_map`.

While this change makes the tests pass, I expected to see `Deserializer::deserialize_enum` called at some point, and it is not. It's possible I'm missing some internal serde magic that causes `deserialize_map` to produce the appropriate enum variant.

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -152,7 +152,36 @@ impl<'de> de::Deserializer<'de> for Deserializer {
     where
         V: Visitor<'de>,
     {
-        vis.visit_str(self.0.value())
+        let v = self.0.value();
+        if v.is_empty() {
```
Comment (wyatt-herkamp) [inline src/de.rs:156]:
```rust
    #[test]
    fn test_from_env_internally_enum_numeric_string() {
        temp_env::with_vars(
            vec![("FOO_TYPE", Some("Y")), ("FOO_BAR", Some("123"))],
            || {
                let t: InternallyEnumStruct = from_env().expect("must success");
                assert_eq!(
                    t.foo,
                    InternallyEnum::Y(EnumNewtype {
                        bar: "123".to_string()
                    })
                )
            },
        )
    }
```
A test like this fails. Basically, strings that look like numbers ```must success: Error("invalid type: integer `123`, expected a string")```

And it seems to have a lot of weird edge cases where it fails.

Comment (trxcllnt) [reply src/de.rs:156]: Yes, that is a consequence of attempting to parse the value numerically before falling back to treating the value as a string. If we don't try to parse the string to a number, it fails in the inverse way -- deserializing a numerical value will fail with `invalid type: string "123", expected an integer`.

I recognize this is a limitation in `serde` that we are working around, so there aren't going to be any good solutions. I'm fine not trying to parse the value as a numeric type first, I only included that to make the existing test pass without modifications.

I suspect the only way a user could handle this is by implementing Deserialize and explicitly specifying the value types in `Visitor::visit_map`. I think that's an acceptable solution, assuming it's documented, because it gives control back to the user, even if the workaround is less ergonomic than we'd like.

As it stands, the choice for users is either to refactor their enums to all be externally-tagged (which isn't always possible, or is possible but would be a breaking change for their users), or to not use `serde-env`.

This PR would give them a third option: use `serde-env` to deserialize their existing API, with the caveat that they may have to implement Deserialize in order to work around a `serde` limitation.
